hdfs-dataset-sink/src/main/java/org/springframework/cloud/stream/module/dataset/sink/DatasetSink.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.dataset.sink;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.messaging.Sink;
+import org.springframework.data.hadoop.store.dataset.DatasetOperations;
+import org.springframework.integration.annotation.ServiceActivator;
+import org.springframework.messaging.Message;
+
+import java.util.Collection;
+import java.util.Collections;
+
+/**
+ * A sink that can be used to insert data into HDFS.
+ *
+ * @author Thomas Risberg
+ */
+@EnableBinding(Sink.class)
+public class DatasetSink {
+
+	private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+	private DatasetOperations datasetOperations;
+
+	@Autowired
+	public void setDataStoreWriter(DatasetOperations datasetOperations) {
+		this.datasetOperations = datasetOperations;
+	}
+
+	@ServiceActivator(inputChannel="toSink")
+	public void datasetSink(Message<?> message) {
```

Comment:
I'd avoid this class in favor of simple `@Bean` for an inline `MessageHandler` impl.
With this `@ServiceActivator(inputChannel="toSink")`, of course.
From other side the same can be utilized with the SI Java DSL. Especially that horror around the `AggregatorFactoryBean`...